bpf/trace_common.h

```diff
@@ -91,25 +97,37 @@ static __always_inline unsigned char *bpf_strstr_tp_loop(unsigned char *buf, int
 #endif
 
 static __always_inline tp_info_pid_t *find_parent_trace() {
-    pid_key_t c_tid = {0};
+    trace_key_t t_key = {0};
+
+    task_tid(&t_key.p_key);
+    u64 extra_id = extra_runtime_id();
+    t_key.extra_id = extra_id;
 
-    task_tid(&c_tid);
     int attempts = 0;
 
     do {        
-        tp_info_pid_t *server_tp = bpf_map_lookup_elem(&server_traces, &c_tid);
+        tp_info_pid_t *server_tp = bpf_map_lookup_elem(&server_traces, &t_key);
 
         if (!server_tp) { // not this goroutine running the server request processing
             // Let's find the parent scope
-            pid_key_t *p_tid = (pid_key_t *)bpf_map_lookup_elem(&clone_map, &c_tid);
-            if (p_tid) {
-                // Lookup now to see if the parent was a request
-                c_tid = *p_tid;
+            if (t_key.extra_id) {
```

Comment:
If we have runtime ID, we lookup in the runtime parent table, else we look at the OS level thread information as before.